static/app/views/replays/detail/console/consoleMessage.tsx

```diff
@@ -45,35 +47,56 @@ function MessageFormatter({breadcrumb}: MessageFormatterProps) {
     ? sprintf.parse(message).filter(parsed => Array.isArray(parsed))
     : [];
 
-  // TODO `%c` is console specific, it applies colors to messages
-  // for now we are stripping it as this is potentially risky to implement due to xss
-  const consoleColorPlaceholderIndexes = placeholders
-    .filter(([placeholder]) => placeholder === '%c')
-    .map((_, i) => i);
-
-  // Retrieve message formatting args
-  const messageArgs = args.slice(0, placeholders.length);
-
-  // Filter out args that were for %c
-  for (const colorIndex of consoleColorPlaceholderIndexes) {
-    messageArgs.splice(colorIndex, 1);
+  // Placeholders can only occur in the first argument and only if it is a string.
+  // We can skip the below code and avoid using `sprintf` if there are no placeholders.
+  if (placeholders.length) {
+    // TODO `%c` is console specific, it applies colors to messages
+    // for now we are stripping it as this is potentially risky to implement due to xss
+    const consoleColorPlaceholderIndexes = placeholders
+      .filter(([placeholder]) => placeholder === '%c')
+      .map((_, i) => i);
+
+    // Retrieve message formatting args
+    const messageArgs = args.slice(0, placeholders.length);
+
+    // Filter out args that were for %c
+    for (const colorIndex of consoleColorPlaceholderIndexes) {
+      messageArgs.splice(colorIndex, 1);
+    }
+
+    // Attempt to stringify the rest of the args
+    const restArgs = args.slice(placeholders.length).map(renderString);
+
+    const formattedMessage = isMessageString
+      ? vsprintf(message.replaceAll('%c', ''), messageArgs)
+      : renderString(message);
+
+    logMessage = [formattedMessage, ...restArgs].join(' ').trim();
+  } else if (
+    breadcrumb.message?.includes('[object Object]') ||
+    Array.isArray(message) ||
+    args.length
```

Comment:
I think there's going to be too many cases to account for, instead we should make this the default case (i.e. the `else`), and try to use the else if to target the "error" case.

e.g. we could have `console.log('foo', ['test', 'test'])`